MSAL/test/app/ios/MSALTestAppAuthorityTypeViewController.m

```diff
@@ -0,0 +1,97 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALTestAppAuthorityTypeViewController.h"
+#import "MSALTestAppSettings.h"
+#import "MSALTestAppB2CAuthorityViewController.h"
+
+@interface MSALTestAppAuthorityTypeViewController ()
+
+@end
+
+@implementation MSALTestAppAuthorityTypeViewController
+
++ (instancetype)sharedController
+{
+    static MSALTestAppAuthorityTypeViewController *s_controller = nil;
+    static dispatch_once_t once;
+    
+    dispatch_once(&once, ^{
+        s_controller = [[MSALTestAppAuthorityTypeViewController alloc] init];
+    });
+    
+    return s_controller;
+}
+
+- (id)init
+{
+    self = [self initWithAuthorities:[[MSALTestAppSettings authorityTypes] mutableCopy]];
+    return self;
+}
+
+- (id)initWithAuthorities:(NSMutableArray *) authorityTypes
+{
+    if(self=[super init]){
+        _authorityTypes = authorityTypes;
+    }
+    return self;
+}
+
+#pragma mark -
+#pragma mark Methods for subclasses to override
+
+- (void)viewDidLoad
+{
+    self.navigationItem.title = @"Select Authority Type";
+    [super viewDidLoad];
+}
+
+- (NSInteger)numberOfRows
+{
+    return _authorityTypes.count;
+}
+
+- (NSString *)labelForRow:(NSInteger)row
+{
+    return _authorityTypes[row];
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    (void)tableView;
+    NSInteger selectedIndex = [indexPath indexAtPosition:1];
```

Comment:
nit: I don't know why we use indexAtPosition, indexPath.row and indexPath.section are normally easier to understand